python/ray/serve/_private/controller.py

```diff
@@ -375,6 +384,109 @@ def _update_proxy_nodes(self):
         new_proxy_nodes.add(self._controller_node_id)
         self._proxy_nodes = new_proxy_nodes
 
+    def _emit_deployment_autoscaling_snapshots(self) -> None:
+        """Emit a structured snapshot log per autoscaling-enabled deployment."""
+        try:
+            app_statuses = self.application_state_manager.list_app_statuses()
+            for app_name in app_statuses.keys():
+                deployment_details = (
+                    self.application_state_manager.list_deployment_details(app_name)
+                )
+                for dep_name, details in deployment_details.items():
+                    autoscaling_config = details.deployment_config.autoscaling_config
+                    if not autoscaling_config:
+                        continue
+                    current_replicas = len(getattr(details, "replicas", []) or [])
+                    raw_target = getattr(details, "target_num_replicas", None)
```

Comment:
same comment here `details.target_num_replicas` cannot be None based on pydantic model